### PR TITLE
Update to Koa 2

### PR DESCRIPTION
**Pull request by dantman**
With the recent changes made and the relative simplicity of the task, it was easier to simply re-do the work of updating to koa@2 than it was to simply rebase the work @pgherveou did in #20.

### CODE REVIEW

**Comment by pgherveou** [comment]
Cool! Sorry never got time to resubmit my work

**Comment by coderhaoxin** [comment]
@dantman please rebase with master

**Comment by dantman** [comment]
Done

**Comment by ilkkao** [comment]
Would be nice to get this one merged when koa current is 2.0.1

**Comment by coderhaoxin** [comment]
I'll merge this within this week :)
Sorry for the delay 😢

File: index.js [hunk 1]
```diff
@@ -37,48 +37,49 @@ function error(opts) {
...
-            response: this.response,
-            error: err.message,
-            stack: err.stack,
-            status: this.status,
-            code: err.code
-          });
-          break;
-      }
-    }
+  return function error(ctx, next){
```
Comment (coderhaoxin) [inline index.js]: @dantman It's better to use `async function`.

**Comment by coderhaoxin** [comment]
updated at https://github.com/koajs/error/pull/28

**Comment by coderhaoxin** [comment]
land 1594f6dab7011ac90c150e97319e0694017de030

**Comment by coderhaoxin** [comment]
`3.0.0` published !

**Pull request closed without merging**